tests/unit/bond_test.py

```diff
@@ -114,6 +116,117 @@ def test_generate_access_token(self):
         self.assertEqual(datetime.fromtimestamp(self.expires_at_epoch), expires_at)
 
+    def test_get_access_token_from_cache(self):
+        data = {"context": {"user": {"name": self.name}}, 'iat': self.issued_at_epoch}
+        encoded_jwt = jwt.encode(data, 'secret', 'HS256')
+        expires_at_initial = datetime.now().timestamp() + 1
+        
+        fake_token_defaults = {
+            FenceKeys.ACCESS_TOKEN: self.fake_access_token,
+            FenceKeys.ID_TOKEN: encoded_jwt,
+            }
+
+        mock_oauth_adapter = OauthAdapter("foo", "bar", "baz", "qux")
+        mock_oauth_adapter.refresh_access_token = MagicMock(
+            return_value={**fake_token_defaults, **{FenceKeys.EXPIRES_AT: expires_at_initial}}
+        )
+
+        fence_api = self._mock_fence_api(json.dumps({"private_key_id": "asfasdfasdf"}))
+        cache_api = FakeCacheApi()
+        bond = Bond(mock_oauth_adapter,
+                    fence_api,
+                    cache_api,
+                    self.refresh_token_store,
+                    FenceTokenVendingMachine(fence_api, cache_api, self.refresh_token_store,
+                                             mock_oauth_adapter, provider_name,
+                                             FakeFenceTokenStorage()),
+                    provider_name,
+                    "/context/user/name",
+                    {})
+        
+        token = str(uuid.uuid4())
+        self.refresh_token_store.save(user_id=self.user_id,
+                                      refresh_token_str=token,
+                                      issued_at=datetime.fromtimestamp(self.issued_at_epoch),
+                                      username=self.name,
+                                      provider_name=provider_name)
+
+        # verify that the new `expired_at` value is returned with `get_access_token`,
+        # when nothing has been cached yet.
+        access_token, expires_at = bond.get_access_token(self.user_id, refresh_threshold=0)
+        self.assertEqual(self.fake_access_token, access_token)
+        self.assertEqual(datetime.fromtimestamp(expires_at_initial), expires_at)
+
+        expires_at_new = expires_at_initial + 100
+
+        mock_oauth_adapter.refresh_access_token = MagicMock(
+            return_value={**fake_token_defaults, **{FenceKeys.EXPIRES_AT: expires_at_new}}
+        )
+
+        # verify that the new `expires_at` value is returned with `generate_access_token`.
+        access_token, expires_at = bond.generate_access_token(self.user_id)
+        self.assertEqual(self.fake_access_token, access_token)
+        self.assertEqual(datetime.fromtimestamp(expires_at_new), expires_at)
+
+        # verify that the cached `expires_at` value is returned with `get_access_token`
+        access_token, expires_at = bond.get_access_token(self.user_id)
+        self.assertEqual(self.fake_access_token, access_token)
+        self.assertEqual(datetime.fromtimestamp(expires_at_initial), expires_at)
```

Comment:
This looks like it's checking the same thing as before the call to `generate_access_token`. Are you verifying that `generate_access_token` does not affect the cache?

TOL: I think that might be an overly specific test. If you do keep it, I think it would be helpful to be explicit about the behavior you're intending to verify.

Reply:
I see your point that this may be overly specific. Ironically, I put it there to make it clearer to the reader what the "real" test (the call to `get_access_token`) is doing, by providing a point of contrast. But it seems like I've done the opposite.

I think it's worth having a quick conversation about this